src/ipc/ipc4/handler.c

```diff
@@ -72,9 +72,9 @@ struct ipc4_msg_data {
 static struct ipc4_msg_data msg_data;
 
 /* fw sends a fw ipc message to send the status of the last host ipc message */
-static struct ipc_msg msg_reply = {0, 0, 0, 0, LIST_INIT(msg_reply.list)};
+static struct ipc_msg msg_reply = {0, 0, 0, 0, false, LIST_INIT(msg_reply.list)};
```

Comment:
are those static structures ever cleaned? If not, once is_shared is set to  true  it'll never be set to back false

Reply:
Thanks, that was a good catch. It seems to me that I managed to cover all cases now.